pytest_splunk_addon/standard_lib/cim_tests/test_templates.py

```diff
@@ -482,3 +482,22 @@ def test_eventtype_mapped_multiple_cim_datamodel(
             f"\nQuery result greater than 0.\nsearch=\n{search} \n \n"
             f"Event type which associated with multiple data model \n{result_str}"
         )
+
+    @pytest.mark.splunk_searchtime_cim
+    @pytest.mark.splunk_requirements
+    @pytest.mark.splunk_requirements_unit
+    def test_cim_fields_recommended(
+        self, splunk_dm_recommended_fields, splunk_searchtime_cim_fields_recommended
+    ):
+        datamodel = splunk_searchtime_cim_fields_recommended["datamodel"]
+        datasets = splunk_searchtime_cim_fields_recommended["datasets"]
+        fields = splunk_searchtime_cim_fields_recommended["fields"]
+
+        fields_from_splunk = splunk_dm_recommended_fields(datamodel, datasets)
+        model_key = f"{datamodel}:{':'.join(datasets)}".strip(":")
+
+        model_fields = fields_from_splunk[model_key]
+
+        assert all(
+            item in set(fields) for item in set(model_fields)
+        ), f"Not all fields from datamodel - {model_fields} found for event definition {fields}"
```

Comment:
I do agree with @artemrys 
Moreover, I'd suggest to change the approach toward using just Data Sets (no Data Models), so that CIM addon can be used in some long term. As an example, in SPL, reference to image_id in Instance Changes (https://docs.splunk.com/Documentation/CIM/5.0.1/User/Change) is: All_Changes.Instance_Changes.image_id

Reply:
Changed log to contain fields, which are defined in datamodel but not in event definition

Reply:
@hsekowski-splunk do you mean change in xml event format to be per dataset? or change the way fields are being extracted from Splunk API?